<launch>
    <!-- No namespace here as we will share this description.
        Access with slash at the beginning -->
    <!--param name="robot_description"
        command="$(find xacro)/xacro.py $(find turtlebot_description)/urdf/turtlebot.urdf.xacro" /-->

    <!-- BEGIN ROBOT 0-->
    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>

    <group ns="robot_0">
        <param name="tf_prefix" value="robot_0" />
        <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
        <include file="$(find turtlebot_sim)/launch/one_robot.launch" >
            <arg name="init_pose" value="-x 0 -y 0 -z 0" />
            <arg name="robot_name"  value="robot_0" />
        </include>
    </group>

    <!-- BEGIN ROBOT 1-->
    <group ns="robot_1">
        <param name="tf_prefix" value="robot_1" />
        <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
        <include file="$(find turtlebot_sim)/launch/one_robot.launch" >
            <arg name="init_pose" value="-x -1.5 -y 0.5 -z 0" />
            <arg name="robot_name"  value="robot_1" />
        </include>
    </group>

    <!-- BEGIN ROBOT 2-->
    <group ns="robot_2">
        <param name="tf_prefix" value="robot_2" />
        <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
        <include file="$(find turtlebot_sim)/launch/one_robot.launch" >
            <arg name="init_pose" value="-x -1.5 -y -0.5 -z 0" />
            <arg name="robot_name"  value="robot_2" />
        </include>
    </group>
</launch>